docs/adr/0009_versioning_strategy.md

```diff
@@ -0,0 +1,299 @@
+# ADR 0009 — Versioning and Release Branching for the OCM Monorepo
+
+* Status: Proposed
+* Deciders: Gergely Brautigam, Fabian Burth, Jakob Moeller, Gerald Morrison, Ilya Khandamirov, Piotr Janik, Matthias Bruns
+* Date: 2025-09-15
+
+## Context
+
+The repository contains multiple independently developed components, e.g., `ocm-cli`, `ocm-controller`, and a root/wrapper component `ocm` (final name to be decided. Acts as a root component referencing versions of the sub-components).
+
+We require a single, operational specification that defines how components are versioned, how release branches are created and maintained, and how CI publishes releases.
+
+Architectural requirement: deciding on a versioning strategy must include the implications for branching. Without an agreed branching model the versioning choice is incomplete and cannot be implemented safely.
+
+## Decision Drivers
+
+* Reproducible, auditable published releases.
+* Maintain per-component autonomy where practical.
+* Support straightforward hotfix and backport workflows for older minor releases.
+* Provide a machine-readable snapshot for documentation and website consumption.
+* Keep CI complexity manageable and deterministic.
+
+## Decision
+
+We adopt a combined, opinionated strategy:
+
+* Versioning: Hybrid approach — each component maintains a local `VERSION` file; the repository also maintains a generated `/VERSIONS.yaml` that mirrors the last published versions of all components (aka snapshot)
+* Branching: Component-scoped persistent release branches — branches follow the pattern `releases/<component>/v<major>.<minor>` and act as maintenance lines for each active minor release. Patches and hotfixes are applied via PRs to these branches.
+* Publishing: An annotated tag `<component>/v<major>.<minor>.<patch>` is the authoritative record of a published artifact.
```

Comment:
you mean of the source of an artifact right?

Reply:
correct